File: .pub-cache/hosted/pub.dev/zego_express_engine-3.23.0/windows/ZegoCustomAudioProcessManager.cpp
#include "include/zego_express_engine/ZegoCustomAudioProcessManager.h"

static std::shared_ptr<ZegoCustomAudioProcessManager> instance_ = nullptr;
static std::once_flag singletonFlag;

std::shared_ptr<ZegoCustomAudioProcessManager> ZegoCustomAudioProcessManager::getInstance() {
    std::call_once(singletonFlag, [&] {
        instance_ = std::make_shared<ZegoCustomAudioProcessManager>();
    });
    return instance_;
}

std::shared_ptr<IZegoFlutterCustomAudioProcessHandler> ZegoCustomAudioProcessManager::getHandler() {
    return handler_;
}

void ZegoCustomAudioProcessManager::setCustomAudioProcessHandler(std::shared_ptr<IZegoFlutterCustomAudioProcessHandler> handler) {
    handler_ = handler;
}   

File: .pub-cache/hosted/pub.dev/zego_express_engine-3.23.0/android/src/main/java/im/zego/zego_express_engine/ZegoCustomAudioProcessManager.java
package im.zego.zego_express_engine;

public class ZegoCustomAudioProcessManager {
    private volatile static ZegoCustomAudioProcessManager singleton;

    private static IZegoFlutterCustomAudioProcessHandler mHander;

    public ZegoCustomAudioProcessManager() {
    }

    public static synchronized ZegoCustomAudioProcessManager getInstance() {
        if (singleton == null) {
            singleton = new ZegoCustomAudioProcessManager();
        }
        return singleton;
    }


    public void setCustomAudioProcessHandler(IZegoFlutterCustomAudioProcessHandler handler) {
        mHander = handler;
    }

    public IZegoFlutterCustomAudioProcessHandler getHandler() {
        return mHander;
    }

}
